era5cli/fetch.py

```diff
@@ -179,8 +180,12 @@ def fetch(self, dryrun=False):
         self._extension()
         # define fetch call depending on split argument
         if not self.merge:
-            # split by variable and year
-            self._split_variable_yr()
+            # split request. Land and ensable requests need to be split by variable,
+            #     year and month to avoid too-large-request error
+            if (self.ensemble or self.land) and self.period == "hourly":
+                self._split_variable_yr_month()
+            else:
+                self._split_variable_yr()
```

Comment:
I wonder if we should make this behavior uniform across all datasets, and/or introduce a flag that controls this?

Reply:
That should be easy enough to add, and ties in well with your general comment to add a flag for splitting requests by month.